src/cascadia/TerminalApp/TerminalWindow.cpp

```diff
@@ -0,0 +1,1262 @@
+            _root->SetInboundListener(true);
+        }
+
+        return _root->Initialize(hwnd);
+    }
+    // Method Description:
+    // - Called around the codebase to discover if this is a UWP where we need to turn off specific settings.
+    // Arguments:
+    // - <none> - reports internal state
+    // Return Value:
+    // - True if UWP, false otherwise.
+    bool TerminalWindow::IsUwp() const noexcept
+    {
+        // use C++11 magic statics to make sure we only do this once.
+        // This won't change over the lifetime of the application
+
+        static const auto isUwp = []() {
+            // *** THIS IS A SINGLETON ***
+            auto result = false;
+
+            // GH#2455 - Make sure to try/catch calls to Application::Current,
+            // because that _won't_ be an instance of TerminalApp::App in the
+            // LocalTests
+            try
+            {
+                result = ::winrt::Windows::UI::Xaml::Application::Current().as<::winrt::TerminalApp::App>().Logic().IsUwp();
+            }
+            CATCH_LOG();
+            return result;
+        }();
+
+        return isUwp;
+    }
+
+    // Method Description:
+    // - Called around the codebase to discover if Terminal is running elevated
+    // Arguments:
+    // - <none> - reports internal state
+    // Return Value:
+    // - True if elevated, false otherwise.
+    bool TerminalWindow::IsElevated() const noexcept
+    {
+        // use C++11 magic statics to make sure we only do this once.
+        // This won't change over the lifetime of the application
+
+        static const auto isElevated = []() {
+            // *** THIS IS A SINGLETON ***
+            auto result = false;
+
+            // GH#2455 - Make sure to try/catch calls to Application::Current,
+            // because that _won't_ be an instance of TerminalApp::App in the
+            // LocalTests
+            try
+            {
+                result = ::winrt::Windows::UI::Xaml::Application::Current().as<::winrt::TerminalApp::App>().Logic().IsElevated();
+            }
+            CATCH_LOG();
+            return result;
+        }();
+
+        return isElevated;
+    }
+
+    // Method Description:
+    // - Build the UI for the terminal app. Before this method is called, it
+    //   should not be assumed that the TerminalApp is usable. The Settings
+    //   should be loaded before this is called, either with LoadSettings or
+    //   GetLaunchDimensions (which will call LoadSettings)
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - <none>
+    void TerminalWindow::Create()
+    {
+        _root->DialogPresenter(*this);
+
+        // Pay attention, that even if some command line arguments were parsed (like launch mode),
+        // we will not use the startup actions from settings.
+        // While this simplifies the logic, we might want to reconsider this behavior in the future.
+        if (!_hasCommandLineArguments && _gotSettingsStartupActions)

```

Comment:
`Create` and `Initialize` both act on the startup actions; is that safe?

Reply:
I mean, yea? Maybe I should rename `Create` to `SetupUI` or something, to differentiate that `Initialize` comes first, then you create the UI